src/transformers/models/rag/modeling_tf_rag.py

```diff
@@ -0,0 +1,1828 @@
+            config=self.config,
+            input_ids=input_ids,
+            attention_mask=attention_mask,
+            decoder_input_ids=decoder_input_ids,
+            decoder_attention_mask=decoder_attention_mask,
+            encoder_outputs=encoder_outputs,
+            past_key_values=past_key_values,
+            doc_scores=doc_scores,
+            context_input_ids=context_input_ids,
+            context_attention_mask=context_attention_mask,
+            use_cache=use_cache,
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            output_retrieved=output_retrieved,
+            return_dict=return_dict,
+            n_docs=n_docs,
+            training=training,
+            kwargs_call=kwargs,
+        )
+
+        # aliasing to minimize code changing
+        input_ids = inputs["input_ids"]
+        attention_mask = inputs["attention_mask"]
+        decoder_input_ids = inputs["decoder_input_ids"]
+        decoder_attention_mask = inputs["decoder_attention_mask"]
+        encoder_outputs = inputs["encoder_outputs"]
+        past_key_values = inputs["past_key_values"]
+        doc_scores = inputs["doc_scores"]
+        context_input_ids = inputs["context_input_ids"]
+        context_attention_mask = inputs["context_attention_mask"]
+
+        use_cache = inputs["use_cache"]
+        output_attentions = inputs["output_attentions"]
+        output_hidden_states = inputs["output_hidden_states"]
+        return_dict = inputs["return_dict"]
+        n_docs = inputs["n_docs"] if inputs["n_docs"] is not None else self.config.n_docs
+        output_retrieved = inputs["output_retrieved"]
+        training = inputs["training"]
+
+        # whether retriever has to be used
+        has_to_retrieve = (
+            self.retriever is not None
+            and (context_input_ids is None or context_attention_mask is None or doc_scores is None)
+            and encoder_outputs is None
+        )
+
+        # encoder_outputs are pre-computed during RAG-token generation
+        if encoder_outputs is None:
+
+            if has_to_retrieve:
+                question_enc_outputs = self.question_encoder(
+                    input_ids, attention_mask=attention_mask, return_dict=True, training=training
+                )
+                # see https://github.com/huggingface/transformers/blob/master/src/transformers/models/dpr/modeling_tf_dpr.py#L91
+                question_encoder_last_hidden_state = question_enc_outputs[
+                    0
+                ]  # hidden states of question encoder => pooler_output
+
+                retriever_outputs = self.retriever(
+                    input_ids,
+                    question_encoder_last_hidden_state.numpy(),  # NEED_HELP : not work in GRAPH mode, tf.make_ndarray doesn't work as well
+                    prefix=self.generator.config.prefix,
+                    n_docs=n_docs,
+                    return_tensors="tf",
+                )
```

Comment:
Might be interesting to investigate to turn the call of the retriever inside a [tf.numpy_function](https://www.tensorflow.org/api_docs/python/tf/numpy_function).

Reply:
I did investigated/tested tf.numpy_function . Unfortunately I could not make it work. However, as explained in the above comment, I have a work-around (to-be PR) on training with `context_input_ids` where this part of code is not executed.

Reply:
What error did you encounter with this function?

The solution with `context_input_ids` looks promising to have a proper model in graph mode as said above 👍